src/etc/inc/interfaces.inc

```diff
@@ -1356,6 +1356,57 @@ function interface_gif_configure(&$gif, $gifkey = "") {
 	return $gifif;
 }
 
+/*
+ * $ipsecifnum = get_ipsecifnum($ikeid, $idx, $create=true);
+ * locates or creates and returns an ipsecifnum in the config.
+ */
+function get_ipsecifnum($ikeid, $idx, $create=true) {
+	global $config;
+	$matches = array_filter($config['ipsec']['vtimaps'], function($val) use($ikeid, $idx){
+              return ($val['reqid']==$ikeid and $val['index']==$idx);
+	});
+	$existing = array_pop($matches);
+	if (isset($existing['ifnum'])) {
+		return $existing['ifnum'];
+	}
+
+	if (!($create)) {
+		return(NULL);
+	}
+	
+	if ((($ikeid < 33) && ($idx < 10)) || (($ikeid < 10) && ($idx < 100))) {
+		$oldformat = "{$ikeid}00{$idx}";
+		$config['ipsec']['vtimaps'][]=array(
+			"reqid" => $ikeid,
+			"index" => $idx,
+			"ifnum" => $oldformat
+		);
+		write_config(gettext("New ipsec vtimap added"));
```

Comment:
I still don't like the idea of this function to change config.  It would be better if the conversion from old format to new format happens on upgrade_config.inc with a version config bump

Reply:
how can I fix this?
there is no upgrade_config.inc in the master branch
I only see this on installed pfSense

Reply:
Yes, there is an upgrade_config.inc on master branch, at:

https://github.com/pfsense/pfsense/blob/master/src/etc/inc/upgrade_config.inc

Reply:
ok, done
now upgrade_201_to_202() converts all IPsec VTI (including disabled) to new <vtimap> format

Reply:
You need to bump `latest_config` at globals.inc to 20.2 otherwise that function will never be called.
https://github.com/pfsense/pfsense/blob/master/src/etc/inc/globals.inc#L73

Also, since upgrade_config.inc will take care of converting it, this block can be removed from `get_ipsecifnum()`

Reply:
this block is used to create new <vtimap> entries
but there is no any mechanism to remove unused <vtimap>
In Progress...

Reply:
All done

Now it correctly updates config.xml when upgrading,
Deletes unused vtimaps after deleting P1/P2 in WebGUI,
Deletes/creates vtimaps on switching from IKEv1/IKEv2 with split connections to IKEv2 and vise-versa

Reply:
Why is it still doing `config_write()` calls in the function (here and once more below)? If it's happening on upgrade, it should be unnecessary to do a write in the function. It should only be done when creating/deleting a VTI, shouldn't it?